src/NuGet.Core/NuGet.Frameworks/FrameworkReducer.cs

```diff
@@ -93,11 +94,38 @@ private NuGetFramework GetNearestInternal(NuGetFramework framework, IEnumerable<
                 // Remove lower versions of compatible frameworks
                 var reduced = ReduceUpwards(compatible);
 
-                // Reduce to the same framework name if possible
-                if (reduced.Count() > 1
-                    && reduced.Any(f => _fwNameComparer.Equals(f, framework)))
+                bool isNet6Era = StringComparer.OrdinalIgnoreCase.Equals(FrameworkConstants.FrameworkIdentifiers.NetCoreApp, framework.Framework) && framework.Version.Major >= 6;
+
+                // Eliminate non-compatible xamarin frameworks
+                if (isNet6Era && framework.HasPlatform && reduced.Any(f => f.Framework.StartsWith("xamarin.", StringComparison.OrdinalIgnoreCase)))
                 {
-                    reduced = reduced.Where(f => _fwNameComparer.Equals(f, framework));
+                    reduced = reduced.Where(f =>
+                    {
+                        string suffix = f.Framework.ToLower(CultureInfo.CurrentCulture).Replace("xamarin.", "");
```

Comment:
Given how IsCompatible works now, do you still need this?

Reply:
oh. Apparently I don't! :)